lib/Dialect/ESI/runtime/cpp/lib/Accelerator.cpp

```diff
@@ -59,8 +136,13 @@ void registerBackend(string name, BackendCreate create) {
 unique_ptr<AcceleratorConnection> connect(Context &ctxt, string backend,
                                           string connection) {
   auto f = internal::backendRegistry.find(backend);
-  if (f == internal::backendRegistry.end())
-    throw runtime_error("Backend not found");
+  if (f == internal::backendRegistry.end()) {
+    // If it's not already found in the registry, try to load it dynamically.
+    loadDynamic(backend);
+    f = internal::backendRegistry.find(backend);
+    if (f == internal::backendRegistry.end())
+      throw runtime_error("Backend not found");
+  }
```

Comment:
A bit in extension of the above; i fear this is some hard-to-debug code, in case there is a naming mismatch _somewhere_ in the whole backend code registration/library naming.

Reply:
I'll add a note to add debug logging when we get a logging framework.